### PR TITLE
DOC Update docs for iodide code removal

### PR DESCRIPTION
**Pull request by dalcde**
Relevant iodide PR to minimize breakage: https://github.com/iodide-project/iodide/pull/3015

### CODE REVIEW

File: docs/changelog.md [hunk 1]
```diff
@@ -1,6 +1,12 @@
 (changelog)=
 # Release notes
 
+## Verison 0.17.0
+*Unreleased*
+
+- Removed iodide-specific code in `pyodide.js`. This breaks compatibility with
+  iodide and other users pretending to be iodide.
```
Comment (rth) [inline docs/changelog.md:8]: I think

```suggestion
  iodide.
```
would be fine. Could we maybe add a badge "Breaking change" in the beginning as in https://scikit-learn.org/stable/whats_new/v0.24.html ?

**Review by rth — COMMENTED** [review]
There is a few more occurrences;
```
docs/index.rst
26:Pyodide may be used in several ways, including in an Iodide notebook, directly
35:   using_pyodide_from_iodide.md

docs/type_conversions.md
8:  (evaluating a Python cell in Iodide)

docs/loading_packages.md
60:For use in Iodide:
```

Ideally we should find a way to deploy docs on readthedocs for v0.16.1 that would not be overwritten by latest development changes, so that iodide users would still have docs to refer to. We would need to get manainer's right on the RDT project for this though.

**Comment by dalcde** [comment]
I don't know how to do badges, but I made a breaking changes section
like 0.16.0

File: docs/changelog.md [hunk 2]
```diff
@@ -1,6 +1,14 @@
 (changelog)=
 # Release notes
 
+## Verison 0.17.0
+*Unreleased*
+
+### Breaking changes
+
+- Removed iodide-specific code in `pyodide.js`. This breaks compatibility with
+  iodide.
```
Comment (rth) [inline docs/changelog.md:10]: For future PRs, it's also helpful to include a link to the PR in the changelog entries. (I'll fix this one in https://github.com/iodide-project/pyodide/pull/987)

**Pull request merged**